### PR TITLE
Make sticky scroll disappear on the line before the end of the scope. Fixes #156999.

### LINKED ISSUE #156999: Sticky scroll header shows when the rest of the function is out of view
Repro:

Scroll to the bottom of a function such that 1/2 the `}` is in view

image

My expectation here is that the sticky header should stop on the line above the function's closing `}` (ie. on line 1072)

### PR DESCRIPTION
**Pull request by aiday-mar**
Fixes #156999.

### CODE REVIEW

**Comment by aiday-mar** [comment]
I had to update the version of spdlog used as it was throwing the following error in the console:

`Error: The module '\\?\C:\Users\t-aidaym\work\vscode\node_modules\spdlog\build\Release\spdlog.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 103. This version of Node.js requires
NODE_MODULE_VERSION 106`

File: src/vs/editor/contrib/stickyScroll/browser/stickyScroll.ts [hunk 1]
```diff
@@ -130,6 +130,7 @@ class StickyScrollController extends Disposable implements IEditorContribution {
 		} else {
 			this._addOutlineRanges(outlineElement, depth);
 		}
+		console.log('ranges : ', this._ranges);
```
Comment (lramos15) [inline src/vs/editor/contrib/stickyScroll/browser/stickyScroll.ts:133]: Debug statement

File: package.json [hunk 2]
```diff
@@ -78,7 +78,7 @@
     "native-keymap": "3.3.0",
     "native-watchdog": "1.4.0",
     "node-pty": "0.11.0-beta11",
-    "spdlog": "^0.13.0",
+    "spdlog": "^0.13.6",
```
Comment (lramos15) [inline package.json:81]: Why was this package bumped?

**Review by lramos15 — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (lramos15) [reply package.json:81]: > I updated the version of spdlog used as it was throwing the following error in the console:
> 
> `Error: The module '\\?\C:\Users\t-aidaym\work\vscode\node_modules\spdlog\build\Release\spdlog.node' was compiled against a different Node.js version using NODE_MODULE_VERSION 103. This version of Node.js requires NODE_MODULE_VERSION 106`
> 
> I am wondering if I should instead revert back to 0.13.0 from 0.13.6 version of spdlog and downgrade my node version?

Did you `git clean -xfd` then run `yarn` and `yarn watch` again. We just bumped electron

Comment (aiday-mar) [reply package.json:81]: Hi, thanks for commenting. I was getting the following error in my console:

Error: The module '\\?\C:\Users\t-aidaym\work\vscode\node_modules\spdlog\build\Release\spdlog.node' was compiled against a different Node.js version using NODE_MODULE_VERSION 103. This version of Node.js requires NODE_MODULE_VERSION 106

And by searching online, some answers suggested to remove spdlog and reinstall it, which increased it to the latest version.

Comment (aiday-mar) [reply package.json:81]: I see all the checks seem to be failing now, I'll revert back to the older version and investigate further the source of my problem.

Comment (aiday-mar) [reply package.json:81]: I'll try git clean as you suggested.

**Review by lramos15 — APPROVED** [review]

**Pull request merged**